### PR TITLE
Upgrade to Angular 15

### PR DESCRIPTION
**Pull request by coverbeck**
**Description**
Upgrade to Angular 15 and Angular Material 15. 

The ticket title is to upgrade to Angular 16, and the description suggests perhaps even 17. But Angular has a migration tool, which made most of the changes in this PR, and it only upgrades one major version at a time. So this is the first step, and will give us a fallback if upgrading to 16/17 doesn't work.

Most changes made by the Angular Migration tool, but here are some additional notes:

- The Angular Material components we are using have been deprecated and will be removed in Angular Material 17. In preparation for that, they've been renamed `Legacy`, hence most of the changes. I may attempt to migrate to the new components, still on Angular 15, before migration to Angular Material 16.
- The spec files for the unit tests for some reason got fussier about imports being declared.
- The HTML anchor tag does not have a disabled attribute. We were using it before without complaints. I had to change the anchor to a button.

**Review Instructions**
Navigate throughout the UI and ensure no pages seem broken or are behaving differently.

**Issue**
SEAB-6140

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
- [x] Check whether this PR disables tests. If it legitimately needs to disable a test, create a new ticket to re-enable it in a specific milestone.

### CODE REVIEW

File: src/app/docs/docs.component.spec.ts [hunk 1]
```diff
@@ -34,15 +36,15 @@ describe('DocsComponent', () => {
...
+  beforeEach(() => {
+    fixture = TestBed.createComponent(DocsComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
 
-  // it('should create', () => {
-  //   expect(component).toBeTruthy();
-  // });
+  it('should create', () => {
```
Comment (coverbeck) [inline src/app/docs/docs.component.spec.ts:45]: This was commented out; in Angular 15 the tests would throw an error for no tests in the spec file.

File: src/app/shared/register-github-app/register-github-app.component.html [hunk 2]
```diff
@@ -29,16 +29,17 @@
     >.
   </p>
   <div fxLayout="column" fxLayoutAlign="center center" fxLayoutGap="2rem" class="mb-4">
-    <a
+    <button
       mat-raised-button
       type="button"
       class="accent-1-dark-btn mat-elevation-z no-underline"
-      [href]="gitHubAppInstallationLink$ | async"
+      (click)="openGitHubApp()"
```
Comment (coverbeck) [inline src/app/shared/register-github-app/register-github-app.component.html:36]: The anchor tag does not have a disabled attribute (line 39), Angular 15 started error on this, so changed to a button, which doesn't have an href. This covers the changes in the controller as well.

File: src/app/shared/register-github-app/register-github-app.component.ts [hunk 3]
```diff
@@ -7,11 +9,24 @@ import { UserQuery } from '../user/user.query';
...
   @Input() public entryType: string;
+  private gitHubAppInstallationLink: string;
 
-  constructor(private sessionQuery: SessionQuery, private userQuery: UserQuery) {}
+  constructor(private sessionQuery: SessionQuery, private userQuery: UserQuery) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this.gitHubAppInstallationLink$
```
Comment (coverbeck) [inline src/app/shared/register-github-app/register-github-app.component.ts:24]: Needed this for the change to a button and using onClick in the HTML.

File: src/material.scss [hunk 4]
```diff
@@ -20,9 +20,18 @@
 // Include the common styles for Angular Material. We include this here so that you only
 // have to load a single css file for Angular Material in your app.
 // Be sure that you only ever include this mixin once!
-@include mat.core();
+// As of v15 mat.legacy-core no longer includes default typography styles.
```
Comment (coverbeck) [inline src/material.scss:23]: This block created by migration tool. I'm not super familiar with typography, but the pages look the same as before, so leaving it as is.

File: src/app/organizations/collections/remove-collection/remove-collection.component.ts [hunk 5]
```diff
@@ -25,11 +25,20 @@ import { CollectionsQuery } from '../../state/collections.query';
 })
 export class RemoveCollectionDialogComponent {
   public collectionsQueryLoading$: Observable<boolean>;
-  constructor(@Inject(MAT_DIALOG_DATA) public data: CollectionDialogData, private collectionsService: CollectionsService, private collectionsQuery: CollectionsQuery) {
+  constructor(
```
Comment (coverbeck) [inline src/app/organizations/collections/remove-collection/remove-collection.component.ts:28]: Just the auto-formatter doing its thing

File: tsconfig.json [hunk 6]
```diff
@@ -9,7 +9,7 @@
     "moduleResolution": "node",
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
-    "target": "es2020",
+    "target": "ES2022",
```
Comment (svonworl) [inline tsconfig.json:12]: Are we pretty confident that everyone who visits Dockstore will have a browser that supports `ES2022`?
Related: https://compat-table.github.io/compat-table/es2016plus/
(https://github.com/tc39/proposal-regexp-match-indices is an issue for some browsers?)
(I copied this question from the PR that upgraded to Angular 14).

Comment (coverbeck) [reply tsconfig.json]: Fair point, not sure, rolled back to es2020.

**Review by svonworl — APPROVED** [review]

File: tsconfig.json (hunk 6 above)
Comment (denis-yuen) [reply tsconfig.json]: Not sure if I'm reading the chart right, but it looks like ES2021 is as compatible as ES2020. 
Don't know if its worth it

**Review by denis-yuen — APPROVED** [review]
That's a lot of deprecation warnings in SonarCloud

**Comment by denis-yuen** [comment]
> So this is the first step, and will give us a fallback if upgrading to 16/17 doesn't work.

This totally makes sense, so no objection. 

But was curious, so looked it up. It seems that this buys us not active support, but three months of security support (till May). 
https://endoflife.date/angular

The L in LTS is a bit relative https://angular.io/guide/releases#actively-supported-versions

File: tsconfig.json (hunk 6 above)
Comment (coverbeck) [reply tsconfig.json]: > Not sure if I'm reading the chart right, but it looks like ES2021 is as compatible as ES2020. Don't know if its worth it

Agreed, it looks the same, but I think I'll keep it to be safe.

**Comment by coverbeck** [comment]
> That's a lot of deprecation warnings in SonarCloud

All the Angular Material components we have been using have been deprecated, and will be phased out in 17. I wanted to tackle those separately, limiting the scope of this PR to the minimum to upgrade to Angular 15.

**Review by kathy-t — APPROVED** [review]

**Pull request merged**